contracts/modules/TokenExchanger.sol

```diff
@@ -33,139 +35,238 @@ contract TokenExchanger is OnlyOwnerModule {
 
     // Mock token address for ETH
     address constant internal ETH_TOKEN_ADDRESS = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    // Signatures of Paraswap's trade methods
+    bytes4 constant internal MULTISWAP = 0xcbd1603e;
+    bytes4 constant internal BUY = 0xbb2a349b;
 
-    // The address of the KyberNetwork proxy contract
-    address public kyber;
-    // The address of the contract collecting fees for Argent.
-    address public feeCollector;
-    // The Argent fee in 1-per-10000.
-    uint256 public feeRatio;
+    // The address of the Paraswap Proxy contract
+    address public paraswapProxy;
+    // The address of the Paraswap contract
+    address public paraswapSwapper;
+    // The label of the referrer
+    string public referrer;
+    // Authorised exchanges
+    mapping(address => bool) public authorisedExchanges;
 
     event TokenExchanged(address indexed wallet, address srcToken, uint srcAmount, address destToken, uint destAmount);
 
+
+    // *************** Constructor ********************** //
+
     constructor(
         IModuleRegistry _registry,
         IGuardianStorage _guardianStorage,
-        address _kyber,
-        address _feeCollector,
-        uint _feeRatio
+        address _paraswap,
+        string memory _referrer,
+        address[] memory _authorisedExchanges
     )
         BaseModule(_registry, _guardianStorage, NAME)
         public
     {
-        kyber = _kyber;
-        feeCollector = _feeCollector;
-        feeRatio = _feeRatio;
+        paraswapSwapper = _paraswap;
+        paraswapProxy = IAugustusSwapper(_paraswap).getTokenTransferProxy();
+        referrer = _referrer;
+
+        for (uint i = 0; i < _authorisedExchanges.length; i++) {
```

Comment:
I think using a black list instead of a white list would give us more flexibility. When Paraswap adds a new DEX we will want to benefit from it without having to redeploy a new module. That's one of the main benefit of using an aggregator. We should only redeploy a new module when the added DEX does not meet our security requirements.

Reply:
For the record we have decided to keep the (fixed) whitelist as it is the only option that does not place any security/trust assumption on both Argent and Paraswap.